infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/devfile/KubernetesComponentToWorkspaceApplier.java

```diff
@@ -144,15 +149,22 @@ public void apply(
 
     List<PodData> podsData = getPodDatas(componentObjects);
 
-    estimateCommandsMachineName(workspaceConfig, k8sComponent, podsData);
+    Map<String, Map<String, String>> podToContainerAndMachineName = estimateMachineNames(podsData);
+
+    linkCommandsToMachineName(workspaceConfig, k8sComponent, podToContainerAndMachineName);
 
     if (Boolean.TRUE.equals(k8sComponent.getMountSources())) {
       applyProjectsVolumes(podsData, componentObjects);
     }
 
     applyEntrypoints(k8sComponent.getEntrypoints(), componentObjects);
 
-    k8sEnvProvisioner.provision(workspaceConfig, environmentType, componentObjects, emptyMap());
```

Comment:
Adding alias attribute for K8s/OS by pre-creating machine configs with given attribute.